wagtail/tests/test_streamfield.py

```diff
@@ -224,6 +224,29 @@ def test_can_assign_as_list(self):
         self.assertIsInstance(fetched_body[0].value, RichText)
         self.assertEqual(fetched_body[0].value.source, "<h2>hello world</h2>")
 
+    def test_custom_value_class(self):
+        class CustomStreamValue(StreamValue):
+            pass
+
+        body_field = JSONStreamModel._meta.get_field("body")
```

Comment:
I think it would be better to have a dedicated model for this in testapp.models, rather than monkey-patching an existing one - and define some kind of useful method on CustomStreamValue (and test that we can call it) to better match real-world use.